pilot/pkg/model/network.go

```diff
@@ -216,61 +236,62 @@ func (mgr *NetworkManager) resolveHostnameGateways(gatewaySet NetworkGatewaySet)
 				// copy the base gateway to preserve the port/network, but update with the resolved IP
 				resolvedGw := gw
 				resolvedGw.Addr = resolved
-				gatewaySet.Insert(resolvedGw)
+				resolvedGatewaySet.Insert(resolvedGw)
 			}
 		}
 	}
+	return resolvedGatewaySet
 }
 
-func (mgr *NetworkManager) IsMultiNetworkEnabled() bool {
-	if mgr == nil {
+func (gws *NetworkGateways) IsMultiNetworkEnabled() bool {
+	if gws == nil {
 		return false
 	}
-	mgr.mu.RLock()
-	defer mgr.mu.RUnlock()
-	return mgr.multiNetworkEnabled
```

Comment:
@dddddai this was to fix something

Reply:
`multiNetworkEnabled` was needed because [we were deleting unresolved gateways from the `NetworkGatewaySet`](https://github.com/istio/istio/pull/46088/files#diff-0af799e3ed1e301215053a27e717344086e1410e0dbd46c05c25d9979175ce4cL193)

Now we preserve unresolved gateways so we can just check `len(gws.byNetwork)`